src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -236,14 +251,17 @@ impl TimeTrigger {
             let weekday = current.weekday().num_days_from_monday() as i64; // Monday is the first day of the week
             let time = Local.with_ymd_and_hms(year, month, day, 0, 0, 0).unwrap();
             let increment = if modulate { n - week0 % n } else { n };
-            return time + Duration::weeks(increment) - Duration::days(weekday);
+            let dur =
+                try_from!(try_weeks, increment, interval) - try_from!(try_days, weekday, interval);
```

Comment:
At each of these points where we are able to bail and return an error, we need to be specific about what data was out of bounds otherwise it makes it difficult to track down the source of the error for a user.

Reply:
broke these out into individual error types depending on the attempted try